### ISSUE TITLE
Extend ForgeModel with tensor mapping for model-specific input/output handling

### ISSUE THREAD

**Issue by vmilosevic** [tenstorrent/tt-forge-models#199]
Enhance the ForgeModel class to include methods that standardize how inputs and outputs are mapped to and from tensors, depending on the model type (e.g., CNNs like ResNet).
Goal is to make inference side uniform regardless of model that is being used.

Implement the following methods in ForgeModel:

```
def image_to_input(self, image):
    """Convert a PIL image to a model-ready input tensor."""
```

	•	Accept a PIL Image object as input.
	•	Handle any image size — the method should preprocess and resize according to the model’s expected input dimensions.
	•	The preprocessing should follow the logic as in the current load_inputs but should take a user-provided image.

```
def output_to_prediction(self, output):
    """Convert model output tensor to human-readable predictions dictionary."""
```

	•	Map the model’s output tensor to a dictionary of predictions.
	•	For CNN models like ResNet, the result should follow this structure:

```
{
    "label": labels[top1_class_idx],
    "probability": f"{top1_class_prob:.4f}%"
}
```

## Example Usage

```
img = Image.open("cat.jpg")
inputs =loader.image_to_input(img).to(self.device)
with torch.no_grad():
   output = compiled_model(inputs).to("cpu")
   prediction = loader.output_to_prediction(output)

print(prediction)
# {"label": "tabby cat", "probability": "98.34%"}
```

Acceptance Criteria
	•	ForgeModel.image_to_input(image) correctly processes arbitrary-size PIL images to model input tensors.
	•	ForgeModel.output_to_prediction(output) returns dictionary-form predictions with label names and probabilities.
	•	Works for all CNN type model types (e.g., ResNet, Mobilenet..)
	•	We can follow up this with similar methods for other types of models

**Comment by vmilosevic** [comment]
I agree with above comments, I was focusing more on specifics of CNNs and this would be more general approach.

Proposed `get_file` improvements are also great, but maybe it can be tracked as separate effort. 

Batching is maybe also separate topic, how to choose optimal batch size for given hardware/model combination, I suppose in inference server use case we want to optimize for throughput so max batch size is prefered, while in other cases we might optimize for latency with smaller batch. Anyway, input loading in batches should be handled per model and have generic interface for the user.
Inference server already has batch input support, where it can take batch of requests from input queue and send it to a worker.

In any case both input and output handling are on critical path, so this is something we also want to optimize in future for max perf, especially for models that run inference under a second.

**Comment by vmilosevic** [comment]
Linking inference server PR https://github.com/tenstorrent/tt-inference-server/pull/857
It contains copy of model loaders from forge-models for few models with changes I needed, I'm linking it as reference.

**Comment by nvukobratTT** [comment]
Here is the brief doc outlining the main points from this issue:
- https://docs.google.com/document/d/1EL4RiI7ZZFm0PysHj0Rg6rNB1OM3DlWa9xExDSvAgOY/edit?tab=t.0

Also, here is the top-level issue that will track this effort:
- https://github.com/tenstorrent/tt-xla/issues/2061

Please review these on Monday, and I'll see to kick off this effort from Tuesday. 

@kmabeeTT I'm planning to focus one or two engineers from my bring-up team to pick this up for the Vision models. Can you do the same and focus on Language models from your domain? :))

**Comment by kmabeeTT** [comment]
Sorry I totally missed this, a lot to catch up on, thanks for continuing on discussion and not being blocked by my lack of replies! For resources for doing work on I'll have to sync with @AleksKnezevic on this, but potentially our two MCW contractors could pause PCC debug effort.

**Comment by ctr-lelanchelianTT** [comment]
> > Thanks, this works great! Can e have this implemented for following models:
> > 
> > 
> > * [x]       resnet/pytorch[x]      vovnet/pytorch[ ]      efficientnet/pytorch[x]      mobilenetv2/pytorch[ ]      segformer/pytorch[ ]      unet/pytorch[ ]      vit/pytorch
> 
> Yup, on track! :)))
> 
> [@ctr-lelanchelianTT](https://github.com/ctr-lelanchelianTT), please share current state for a given models :))

Hi @nvukobratTT,
Segformer, UNet, and ViT are still pending,I’ll work on them right away.
